gsplat/relocation.py

```diff
@@ -1,20 +1,19 @@
 import math
+from typing import Tuple
+
 import torch
 from torch import Tensor
-from .cuda._wrapper import _make_lazy_cuda_func
 
-N_MAX = 51
-BINOMS = torch.zeros((N_MAX, N_MAX)).float().cuda()
-for n in range(N_MAX):
-    for k in range(n + 1):
-        BINOMS[n, k] = math.comb(n, k)
+from .cuda._wrapper import _make_lazy_cuda_func
 
 
 def compute_relocation(
     opacities: Tensor,  # [N]
     scales: Tensor,  # [N, 3]
     ratios: Tensor,  # [N]
-) -> tuple[Tensor, Tensor]:
+    binoms: Tensor,  # [n_max, n_max]
+    n_max: int,
```

Comment:
@jefequien Would you mind helping me write the doc string for these two? I'm not sure how to best describe them. Dumping your comments here is fine I can add it to the code!

Reply:
binoms: precomputed lookup table for binomial coefficients used in Equation 9 in "3D Gaussian Splatting as Markov Chain Monte Carlo".
n_max: size of lookup table for binomial coefficients. Clamp values larger than this to prevent out-of-bounds exceptions.
```
{'binoms': tensor([[   1,    0,    0,  ...,    0,    0,    0],
        [   1,    1,    0,  ...,    0,    0,    0],
        [   1,    2,    1,  ...,    0,    0,    0],
        ...,
        [   1,   48, 1128,  ...,    1,    0,    0],
        [   1,   49, 1176,  ...,   49,    1,    0],
        [   1,   50, 1225,  ..., 1225,   50,    1]]), 'n_max': 51}
```
You can probably remove n_max and use binoms.size(0) on the cuda side.